Lib/fractions.py

```diff
@@ -244,7 +243,8 @@ def __new__(cls, numerator=0, denominator=None):
                 self._denominator = numerator.denominator
                 return self
 
-            elif isinstance(numerator, (float, Decimal)):
+            elif (isinstance(numerator, numbers.Number) and
+                  hasattr(numerator, 'as_integer_ratio')):
```

Comment:
Sorry, but this looks wrong for me.  While ``as_integer_ratio()`` methods for builtin types have ratio in lowest terms and with a positive denominator, we have no such constraint.  It's possible to create unnormalized fractions, but some Fraction methods work incorrectly for such input:
```pycon
>>> from fractions import Fraction as F
>>> import numbers
>>> class R:
...     def __init__(self, ratio):
...         self._ratio = ratio
...     def as_integer_ratio(self):
...         return self._ratio
... 
>>> numbers.Number.register(R)
<class '__main__.R'>
>>> F(R((6, 4)))
Fraction(6, 4)
>>> F(R((6, 4))) + F(R((2, 2)))
Fraction(5, 2)
>>> 1/F(R((6, 4)))
Fraction(4, 6)
>>> F(R((6, 4))) == F(3, 2)
False
>>> F(R((6, 4))).limit_denominator(3)
Traceback (most recent call last):
  File "<python-input-7>", line 1, in <module>
    F(R((6, 4))).limit_denominator(3)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
  File "/home/sk/src/cpython/Lib/fractions.py", line 397, in limit_denominator
    a = n//d
        ~^^~
ZeroDivisionError: division by zero
```

Please either normalize input or mention requirements for as_integer_ratio() in docs.

Reply:
I prefer to update the documentation. For builtin types calling normalization would be a waste of time.